## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 460330`
Commit: `c00a6c1634012a9b1f7d35199912073386d42959`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>kitty</li>
    <li>kitty.kitten</li>
    <li>kitty.shell_integration</li>
    <li>kitty.terminfo</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 28 packages built:</summary>
  <ul>
    <li>bat-extras.batdiff</li>
    <li>bat-extras.batgrep</li>
    <li>bat-extras.batman</li>
    <li>bat-extras.batpipe</li>
    <li>bat-extras.batwatch</li>
    <li>bat-extras.core</li>
    <li>bat-extras.prettybat</li>
    <li>direnv</li>
    <li>fish</li>
    <li>fish-lsp</li>
    <li>fish.doc</li>
    <li>fishMinimal</li>
    <li>fishMinimal.doc</li>
    <li>fishPlugins.bass</li>
    <li>fishPlugins.done</li>
    <li>fishPlugins.fishtape</li>
    <li>fishPlugins.fishtape_3</li>
    <li>fishPlugins.fzf-fish</li>
    <li>fishPlugins.humantime-fish</li>
    <li>fishPlugins.pure</li>
    <li>fzf-git-sh</li>
    <li>mise</li>
    <li>nixpkgs-manual</li>
    <li>oh-my-fish</li>
    <li>swaynotificationcenter</li>
    <li>tests.writers.bin.fish</li>
    <li>tests.writers.simple.fish</li>
    <li>vimPlugins.direnv-vim</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>kitty</summary>
<pre>    end_time = 499882.467813835
    msg = &#x27;The condition was not met&#x27;
    q = &lt;function ShellIntegration.test_fish_integration.&lt;locals&gt;.redrawn at 0x7fffcb236980&gt;
    self = &lt;kitty_tests.PTY object at 0x7fffcb0614f0&gt;
    timeout = 10
    timeout_msg = None
TimeoutError: Timed out after 10 seconds: The condition was not met. Screen contents as repr:
&#x27;[?200_test_comp_path              &lt;right\nok\nleft&gt;echo $COLUMNS                &lt;right\n\n\n\n\n\n\n\n\n\n\n\n\n&#x27;
Screen contents:
[?200_test_comp_path              &lt;right
ok
left&gt;echo $COLUMNS                &lt;right

----------------------------------------------------------------------
Ran 158 tests in 32.842s

FAILED (errors=2, skipped=2)
All Go tests succeeded, ran in 33.1 seconds
Error: Some tests failed!
make: *** [Makefile:16: test] Error 1</pre>
</details>
</details>

--- a/nixos/modules/programs/fish.nix
+++ b/nixos/modules/programs/fish.nix
@@ -259,7 +259,6 @@ in
               name = "fish_patched-completion-generator";
               srcs = [
                 "${cfg.package}/share/fish/tools/create_manpage_completions.py"
-                "${cfg.package}/share/fish/tools/deroff.py"
               ];
               unpackCmd = "cp $curSrc $(basename $curSrc)";
               sourceRoot = ".";

--- a/pkgs/by-name/fi/fish/nix-darwin-path.patch
+++ b/pkgs/by-name/fi/fish/nix-darwin-path.patch
@@ -1,12 +1,12 @@
 diff --git a/share/config.fish b/share/config.fish
-index d85fd1e185..c564e45b27 100644
+index 73148ac25..1964e30be 100644
 --- a/share/config.fish
 +++ b/share/config.fish
-@@ -158,6 +158,7 @@
+@@ -175,6 +175,7 @@ and __fish_set_locale
  #
  if status --is-login
      if command -sq /usr/libexec/path_helper
 +        and not set -q __NIX_DARWIN_SET_ENVIRONMENT_DONE
-         # Adapt construct_path from the macOS /usr/libexec/path_helper
-         # executable for fish; see
-         # https://opensource.apple.com/source/shell_cmds/shell_cmds-203/path_helper/path_helper.c.auto.html .
+         __fish_macos_set_env PATH /etc/paths '/etc/paths.d'
+         if test -n "$MANPATH"
+             __fish_macos_set_env MANPATH /etc/manpaths '/etc/manpaths.d'

--- a/pkgs/by-name/fi/fish/package.nix
+++ b/pkgs/by-name/fi/fish/package.nix
@@ -7,7 +7,6 @@
   glibcLocales,
   gnused,
   gnugrep,
-  groff,
   gawk,
   man-db,
   ninja,
@@ -152,13 +151,13 @@ let
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "fish";
-  version = "4.1.2";
+  version = "4.2.0";
 
   src = fetchFromGitHub {
     owner = "fish-shell";
     repo = "fish-shell";
     tag = finalAttrs.version;
-    hash = "sha256-oNRC1NWYE0LEK2a/7nHtlmp20f8hn/1FZgaySqzwSbg=";
+    hash = "sha256-t5whU+byERJ+nDLigJ5IznvEg3MUsVqhpGdWFzF+T4Q=";
   };
 
   env = {
@@ -169,7 +168,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   cargoDeps = rustPlatform.fetchCargoVendor {
     inherit (finalAttrs) src patches;
-    hash = "sha256-7mYWCHH6DBWTIJV8GPRjjf6QulwlYjwv0slablDvBF8=";
+    hash = "sha256-4pDbD7hetN7wGdPr2csgVWsqtYKMj6jpYm7zBKra+bU=";
   };
 
   patches = [
@@ -192,13 +191,15 @@ stdenv.mkDerivation (finalAttrs: {
 
   # Fix FHS paths in tests
   postPatch = ''
-    substituteInPlace src/builtins/tests/test_tests.rs \
+    substituteInPlace src/builtins/test.rs \
       --replace-fail '"/bin/ls"' '"${lib.getExe' coreutils "ls"}"'
 
-    substituteInPlace src/highlight/tests.rs \
+    substituteInPlace src/highlight/highlight.rs \
       --replace-fail '"/bin/echo"' '"${lib.getExe' coreutils "echo"}"' \
       --replace-fail '"/bin/c"' '"${lib.getExe' coreutils "c"}"' \
-      --replace-fail '"/bin/ca"' '"${lib.getExe' coreutils "ca"}"' \
+      --replace-fail '"/bin/ca"' '"${lib.getExe' coreutils "ca"}"'
+
+    substituteInPlace src/highlight/file_tester.rs \
       --replace-fail '/usr' '/'
 
     substituteInPlace tests/checks/cd.fish \
@@ -296,7 +297,6 @@ stdenv.mkDerivation (finalAttrs: {
     coreutils
     gnugrep
     gnused
-    groff
     gettext
   ]
   ++ lib.optional (!stdenv.hostPlatform.isDarwin) man-db;
@@ -341,9 +341,6 @@ stdenv.mkDerivation (finalAttrs: {
   postInstall = ''
     substituteInPlace "$out/share/fish/functions/grep.fish" \
       --replace-fail "command grep" "command ${lib.getExe gnugrep}"
-
-    substituteInPlace "$out/share/fish/functions/__fish_print_help.fish" \
-      --replace-fail "nroff" "${lib.getExe' groff "nroff"}"
 
     substituteInPlace $out/share/fish/completions/{sudo.fish,doas.fish} \
       --replace-fail "/usr/local/sbin /sbin /usr/sbin" ""

--- a/pkgs/by-name/fi/fish/package.nix
+++ b/pkgs/by-name/fi/fish/package.nix
@@ -383,6 +383,7 @@ stdenv.mkDerivation (finalAttrs: {
       winter
       sigmasquadron
       rvdp
+      lonerOrz
     ];
     mainProgram = "fish";
   };

